docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,348 @@
+
+###### New behaviour
+
+I think it makes sense both on sidecar and egress.
+If we're troubleshooting something it would be great to see spans from sidecar going to ExternalService.
+
+##### MeshTrafficPermission
+
+###### Old ExternalService behaviour
+
+Both MeshTrafficPermission and TrafficPermission only works with egress.
+
+###### New behaviour
+
+It is already implemented on egress for MeshTrafficPermission.
+
+If on the sidecar:
+- could you circumvent this limitation by trying to directly call through egress?
+
+##### MeshLoadBalancingStrategy
+
+###### Old ExternalService behaviour
+
+There is no old behaviour because there was no concept of Endpoints in External Service
+
+###### New behaviour
+
+Only the `loadBalancer` field makes sense in this regard.
+
+If on sidecar:
+- if you don't need/want egress it still makes sense to pick load balancing mechanism
+
+If on egress:
+- makes the same sense on as on sidecar
+
+##### MeshTCPRoute and MeshHTTPRoute
+
+###### Old ExternalService behaviour
+
+It worked for both egress and sidecar.
+
+###### New behaviour
+
+I don't see why we would change that behaviour.
+
+#### For outbound policies on the sidecar, for inbound policies on egress
+
+It naturally fits our targetRef mechanism.
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### `to`
+
+In `to` section we want to capture all the traffic going `to` a `MeshExternalService.
+
+The situation is pretty straightforward:
+
+```yaml
+kind: MeshTimeout
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: sidecar # just to be explicit that it's on the sidecar? see section 'Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense'
+  to:
+  - targetRef:
+      kind: MeshExternalService
+    default:
+      idleTimeout: 20s
+```
+
+###### `from`
+
+It never makes sense to put `MeshExternalService` in `from` section because traffic never comes from `MeshExternalService`.
+
+###### Top level
+
+We want to have the option to configure inbound policies for `MeshExternalService`.
+
+One option is to use `kind: Mesh`, `proxyType` + `sectionName` just like in [MeshService](https://github.com/kumahq/kuma/blob/7e69b872404251d4e5cd766225fdd1b821cc6c5d/docs/madr/decisions/046-meshservice-policy-matching.md?plain=1#L192):
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: egress # when it's egress it's MES name
+    sectionName: httpbin
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        backends:
+          - type: File
+            file:
+              path: /tmp/access.log
+```
```

Comment:
Isn't just
```
kind: MeshAccessLog
spec:
  targetRef:
    kind: Mesh
    proxyType: egress # or just kind: ZoneEgress
  to:
    - targetRef:
        kind: MeshExternalService
        name: httpbin
      default:
        backends:
          - type: File
            file:
              path: /tmp/access.log
```
?

It's configured on one filter chain anyways. Basically what Ilya suggested

Reply:
I think MAL is not a great example here - MeshTrafficPermission is better. There is no `to` in MTP and you still want to select which services can reach which MES.

Reply:
```
apiVersion: kuma.io/v1alpha1
kind: MeshTrafficPermission
metadata:
  namespace: kuma-system
  name: allow-orders
spec:
  targetRef:
    kind: Mesh
    proxyType: egress
    sectionName: httpbin # we need this field to know which MES to allow traffic from proxies in `from` section
  from:
    - targetRef:
        kind: MeshSubset
        tags:
           kuma.io/service: orders
      default:
        action: Allow
```

If we wanted to use "to" in MTP:

```
apiVersion: kuma.io/v1alpha1
kind: MeshTrafficPermission
metadata:
  namespace: kuma-system
  name: allow-orders
spec:
  targetRef:
    kind: Mesh
    proxyType: egress
  to:
    - targetRef:
        kind: MeshExternalService
        name: httpbin
      default:
        action: Allow
```

we have a problem that we don't know which services are allowed to talk to this MES (`orders` in previous). This config does not make it possible.

Reply:
talked about offline